src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -194,6 +201,10 @@ public function denormalize($data, $class, $format = null, array $context = arra
             }
         }
 
+        if (!empty($extraAttributes)) {
+            throw new ExtraAttributesException(sprintf('Extra attributes are not allowed ("%s" are unknown).', implode('", "', $extraAttributes)));
```

Comment:
It makes stricter exception API. It is harder to misuse it, it decreases possible inconsistency between messages in different places if an exception has many usages in the code. It makes easier to unit test the exception if the message-forming logic becomes tricky.